astroquery/query.py

```diff
@@ -10,20 +10,34 @@
 import io
 import os
 import requests
+import email.utils as eut
+from datetime import datetime
+from time import mktime
 
 import six
 from astropy.config import paths
 from astropy.logger import log
 import astropy.units as u
 from astropy.utils.console import ProgressBarOrSpinner
 import astropy.utils.data
+from requests import Session
+import logging
+import time
 
 from . import version
 from .utils import system_tools
 
 __all__ = ['BaseQuery', 'QueryWithLogin']
 
 
+# Session retries constants
+MAX_RETRY_DELAY = 128  # maximum delay between retries
+# start delay between retries when Try_After not sent by server.
+DEFAULT_RETRY_DELAY = 30
+MAX_NUM_RETRIES = 6
```

Comment:
I would like to see @keflavich and/or @ceb8  opinion on this, whether these should be configurable on the module level, or keep them hardwired here.

Reply:
I'd lean toward making them configurable.